src/index.js

```diff
@@ -171,49 +171,65 @@ export default class IndexedCache {
     // Once the assets have been fetched, apply them synchronously. Since
     // the time take to execute a script is not guaranteed, use the onload() event
     // of each element to load the next element.
-    await Promise.all(promises).then((results) => {
-      results.forEach((r, n) => {
+    await Promise.allSettled(promises).then((results) => {
+      results.forEach((cur, n) => {
+        if (cur.status === 'rejected') {
+          this._applyElement(objs[n])
+          return
+        }
+
         if (n >= results.length - 1) {
           return
         }
 
-        r.obj.el.onload = () => {
-          this._applyBlob(results[n + 1].obj, results[n + 1].data.blob)
+        cur.value.obj.el.onload = cur.value.obj.el.onerror = () => {
+          this._applyElement(results[n + 1].value.obj, results[n + 1].value.data.blob)
         }
       })
 
       // Start the chain by loading the first element.
-      this._applyBlob(results[0].obj, results[0].data.blob)
+      this._applyElement(results[0].value.obj, results[0].value.data.blob)
     })
 
     return objs
   }
 
-  async _loadObject (obj) {
+  // Get the object from the DB and if that fails, fetch() it over HTTP
+  // This function should not reject a promise and in the case of failure,
+  // will return a dummy data object as if it were fetched from the DB.
+  async _getObject (obj) {
```

Comment:
I see few methods are marked as `async` even though no `await` is not used, is there any reason?

Reply:
They needn't be anymore. Will clean up in this branch.